pkg/controller/ingress/ingress_controller_reconciler_test.go

```diff
@@ -40,21 +56,60 @@ func TestIngressControllerReconcilerIntegration(t *testing.T) {
 	}
 
 	// Create resource
+	ingressCn := "testing.webapprouting.kubernetes.azure.com"
 	i := &IngressControllerReconciler{
-		client:    c,
-		resources: []client.Object{obj},
-		logger:    logr.Discard(),
+		client:      c,
+		resources:   []client.Object{obj},
+		logger:      logr.Discard(),
+		ingInformer: ingInformer,
+		className:   ingressCn,
 	}
 	require.NoError(t, i.tick(context.Background()))
 
-	// Prove the resource exists
+	// Prove the resource doesn't exist yet
 	actual := &corev1.Namespace{}
+	require.True(t, errors.IsNotFound(c.Get(context.Background(), client.ObjectKeyFromObject(obj), actual)), "expected not found error")
```

Comment:
can we abstract the expect to exist/ not exist into functions to improve readability instead of having many copies of the same long line with the same comment